### PR TITLE
Sync: Add checksum for options

### PR DESCRIPTION
**Pull request by lezama**
Add checksum for options to `options_checksum`

cc @gravityrail

### CODE REVIEW

File: tests/php/sync/server/class.jetpack-sync-test-replicastore.php [hunk 1]
```diff
@@ -174,6 +174,13 @@ function delete_option( $option ) {
 		$this->options[ $option ] = false;
 	}
 
+	function options_checksum() {
+		return strtoupper( dechex( array_reduce( Jetpack_Sync_Defaults::$default_options_whitelist, array( $this, 'option_checksum' ), 0 ) ) );
+	}
+
+	private function option_checksum( $carry, $option_name ) {
```
Comment (gravityrail) [inline tests/php/sync/server/class.jetpack-sync-test-replicastore.php]: you need to check `isset( $this->options[ $option_name ] )`

Comment (lezama) [inline tests/php/sync/server/class.jetpack-sync-test-replicastore.php]: done!

File: tests/php/sync/server/class.jetpack-sync-test-replicastore.php [hunk 2]
```diff
@@ -46,11 +46,18 @@ function reset() {
...
 		$this->reset();
 	}
-	
+
 	function full_sync_end( $checksum ) {
 		// noop right now
 	}
 
+	function get_wp_version() {
+		return $this->wp_version;
```
Comment (enejb) [inline tests/php/sync/server/class.jetpack-sync-test-replicastore.php]: I think we don't need this any more.

**Comment by jeherve** [comment]
@lezama Is this still in progress, or can it be reviewed? If so, could you rebase?

Thanks!

**Comment by lezama** [comment]
@gravityrail what do you think about this? should we add this or it doesn't add any value?

**Comment by gravityrail** [comment]
@lezama I think it's worth merging this, even if it doesn't turn out to be 100% useful, since we can always choose to ignore it if we need to.

**Pull request merged**